libvirt/tests/src/svirt/dac_nfs_disk.py

```
@@ -78,17 +79,29 @@ def run(test, params, env):
     vm = env.get_vm(vm_name)
     vmxml = VMXML.new_from_inactive_dumpxml(vm_name)
     backup_xml = vmxml.copy()
+    vmosxml = vmxml.os
 
     # Backup domain disk label
     disks = vm.get_disk_devices()
     backup_labels_of_disks = {}
     for disk in disks.values():
         disk_path = disk['source']
-        f = os.open(disk_path, 0)
-        stat_re = os.fstat(f)
-        backup_labels_of_disks[disk_path] = "%s:%s" % (stat_re.st_uid,
-                                                       stat_re.st_gid)
-        os.close(f)
+        label = check_ownership(disk_path)
+        if label:
+            backup_labels_of_disks[disk_path] = label
+
+    try:
+        if vmosxml.nvram:
+            nvram_path = vmosxml.nvram
+            if not os.path.exists(nvram_path):
+                # Need libvirt automatically generate the path
+                vm.start()
+                vm.destroy(gracefully=False)
+            label = check_ownership(nvram_path)
+            if label:
+                backup_labels_of_disks[nvram_path] = label
+    except xcepts.LibvirtXMLNotFoundError:
+        pass
```

I don't think we need try ...except here, which will suppress the error if have since 'pass' is used here.

---

No, we need for line 94

---

Better output some logs  instead of pass if you want to ignore exception.

---

Better output something here instead of pass. Suppose one situation that you expect nvram should be there, but there actually not due to unknown reason